linalg/solvers.hpp

```diff
@@ -169,10 +169,13 @@ class IterativeSolver : public Solver
 
    ///@}
 
-   /// Return the dot product of @a x and @a y
-   double Dot(const Vector &x, const Vector &y) const;
+   /** @brief Return the standard (L2) inner dot product of @a x and @a y
```

Comment:
The "L2" can be misunderstood as the integral L2 inner product. Instead, I suggest we use "(l2, i.e. Euclidean)" here. Also "inner dot product" seems redundant, I'd just use "inner product".